### DISCUSSION TITLE
JWT Access Token expire time validation

### DISCUSSION THREAD

**Discussion by LeoBorai** [saleor/saleor#8998] [Q&A] [answered]
Using Saleor's Demo instance on demo.saleor.io and running this repository locally I noticed that JWT Access Token expiration time is not validated by the server.

If I send a token which `exp` claim is in the past, Saleor API will consume the token anyways without complaining, I expected it to be rejected so I have to refresh the token. Access Tokens default TTL is quite short (5mins) while Refresh Tokens last for a month, so that also makes me understand that I probably have to refresh tokens frequently.

Perhaps I went over some documentation without noticing this? if so I would really appreciate a link so I can clarify my doubts!

## Example

This token:

```
eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpYXQiOjE2NDMwMzk3ODMsIm93bmVyIjoic2FsZW9yIiwiZXhwIjoxNjQzMDQwMDgzLCJ0b2tlbiI6IlJVdkZaVm9kVnVMdCIsImVtYWlsIjoiam9obkB3aWNrLmNvbSIsInR5cGUiOiJhY2Nlc3MiLCJ1c2VyX2lkIjoiVlhObGNqbzVPRGs9IiwiaXNfc3RhZmYiOmZhbHNlfQ.TqYTAxtrBk5Gug5rHvpN1ALQQVLfmcz6Hbm_4NR_jB0
```

Which payload is:

```jwt
{
  "iat": 1643039783,
  "owner": "saleor",
  "exp": 1643040083,
  "token": "RUvFZVodVuLt",
  "email": "john@wick.com",
  "type": "access",
  "user_id": "VXNlcjo5ODk=",
  "is_staff": false
}
```

Refer to: [JWT.io](https://jwt.io/#debugger-io?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpYXQiOjE2NDMwMzk3ODMsIm93bmVyIjoic2FsZW9yIiwiZXhwIjoxNjQzMDQwMDgzLCJ0b2tlbiI6IlJVdkZaVm9kVnVMdCIsImVtYWlsIjoiam9obkB3aWNrLmNvbSIsInR5cGUiOiJhY2Nlc3MiLCJ1c2VyX2lkIjoiVlhObGNqbzVPRGs9IiwiaXNfc3RhZmYiOmZhbHNlfQ.TqYTAxtrBk5Gug5rHvpN1ALQQVLfmcz6Hbm_4NR_jB0)

When provided as variable to:

```graphql
mutation($token: String!) {
  tokenVerify(token: $token) {
    errors {
      field
      code
    }
    isValid
    user {
      email
      isActive
    }
  }
}
```

I get:

```json
{
  "data": {
    "tokenVerify": {
      "errors": [],
      "isValid": true,
      "user": {
        "email": "john@wick.com",
        "isActive": true
      }
    }
  },
  "extensions": {
    "cost": {
      "requestedQueryCost": 0,
      "maximumAvailable": 250
    }
  }
}
```

I expected some `ExpiredToken` -ish error.

**Comment by korycins** [comment, marked as answer]
Hey @EstebanBorai,
Thanks for raising this. 
The expired token will raise an exception when `settings.JWT_EXPIRE` will be set to True. Temporary it is set to False on demo, as storefront deployed on demo instance doesn't support expiration. (Dashboard already supports expirations)
https://docs.saleor.io/docs/3.0/developer/running-saleor/configuration#jwt_expire

**Reply by LeoBorai** [reply]
Thanks so much!
